infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/local/providers/LocalWorkspaceFolderPathProvider.java

```diff
@@ -44,7 +45,7 @@
     public static final String ALLOW_FOLDERS_CREATION_ENV_VARIABLE = "CHE_WORKSPACE_STORAGE_CREATE_FOLDERS";
     public static final String WORKSPACE_STORAGE_PATH_ENV_VARIABLE = "CHE_WORKSPACE_STORAGE";
 
-    private final Provider<WorkspaceManager> workspaceManager;
+    private final WorkspaceDao               workspaceDao;
```

Comment:
I did it because of circular dependency.
`WorkspaceManager` needs `WorkspaceRuntimes`
`WorkspaceRuntimes` needs `DockerRuntimeInfrastructure`
`DockerRuntimeInfrastructure` needs `LocalCheInfrastructureProvisioner`
`LocalCheInfrastructureProvisioner` needs `LocalWorkspaceFolderPathProvider`
`LocalWorkspaceFolderPathProvider` finally needs `WorkspaceManager`

it can't be solved with provider as manager is used in recovery.

Other option would be to initialize `WorkspaceRuntimes` before `WorkspaceManager` but looks a bit ugly, as manager is the one who uses runtimes.

I didn't spend much time trying to get workspace name from some start context, maybe you do know a better way to do it?